doc/clang.txt

```diff
@@ -265,6 +265,7 @@ A: Please use with neocomplete, they can work well together.
         if !exists('g:neocomplete#force_omni_input_patterns')
           let g:neocomplete#force_omni_input_patterns = {}
         endif
+        let g:neocomplete#force_overwrite_completefunc = 1
```

Comment:
> I am planing to figure out how neocomplet ie working with asynchronous (as it's too slow sometimes for large c++ project) complete sources like clang

It is impossible.
If you want to use neocomplete with async, you should use deoplete in neovim.